### DISCUSSION TITLE
Magnesium fire: Conservative simulation

### DISCUSSION THREAD

**Discussion by firestudent** [firemodels/fds#12763] [Q&A]
I am currently conducting a study on the effects of combustion of a pile of magnesium granules on metallic structures.  I know that FDS cannot directly simulate magnesium combustion and I am interested in implementing a **conservative representation** of the combustion process.

Among various publications regarding magnesium combustion, I came across an uncredited data point suggesting that flame heights resulting from magnesium combustion do not exceed 300mm. Could you please provide any credible sources validating this information? If this data were "confirmed," I would like to impose a conservative limit on flame height.

The magnesium pile does not exceed 5kg with a maximum surface area of 1 square meter. Based on the information provided, my idea is to impose a reasonably high Heat Release Rate per Unit Area (HRRPUA) with a contained flame height. 

Thank you for your assistance in advance.

**Comment by rmcdermo** [comment]
Without a citation to the paper, I don't see how we can comment on the credibility of the flame height limit.  But, there must be more to it than "magnesium flames never get above 300 mm".

However, if you know the heat release (mass loss rate * heat of combustion), then you can rather easily use an INIT line with a specified HRRPUV and XB for the volume.  You can RAMP the HRRPUV and mimic the heat generated as you like.  This does not give you all the correct dynamics, but to first order it gives you something reasonable.

**Comment by 19ap** [comment]
Hello!

I’d appreciate some assistance with an issue in my magnesium combustion simulation. I didn’t think it was worth opening a new discussion, so I’ll add the follow-up to this one.
 
I’ve set up the reaction according to section 13.2.2 of the User's Guide, using the "pvc_combustion.fds" example as a base. When I modify the example to simulate an Mg reaction instead of PVC, it runs successfully (with additional adjustments to the MASS_FRACTION in the &INIT line and FIXED_MIX_TIME in the &COMB line).

However, when I attempt a case with a larger mesh and add Mg as a surface layer on top of an &OBST, the simulation terminates after about 1.5 seconds due to Numerical Instability. Before it stops, a warning appears about minimum density in cells near the 'FIRE-Mg' SURF. 

The purpose of the simulation is to track the development of air temperature above the fire and to estimate the temperatures on surfaces surrounding the fuel pan, which are heated by the fire source.

Here is the code:

&HEAD CHID='mg_fire', TITLE='Case of Mg defined fuel'/
&TIME T_END=90.0/
&MESH IJK=40,40,40, XB=-1.0,1.0,-1.0,1.0,0.0,2.0/

&SPEC ID='MAGNESIUM'/
&SPEC ID='NITROGEN', LUMPED_COMPONENT_ONLY=.TRUE./
&SPEC ID='OXYGEN', LUMPED_COMPONENT_ONLY=.TRUE./
&SPEC ID='MAGNESIUM OXIDE', MW=40.3044, ENTHALPY_OF_FORMATION=-601.7, LUMPED_COMPONENT_ONLY=.TRUE./
&SPEC ID='AIR', BACKGROUND=.TRUE., SPEC_ID(1)='NITROGEN', VOLUME_FRACTION(1)=1.88, SPEC_ID(2)='OXYGEN', VOLUME_FRACTION(2)=0.5/
&SPEC ID='PRODUCTS', SPEC_ID(1)='MAGNESIUM OXIDE', VOLUME_FRACTION(1)=1.0, SPEC_ID(2)='NITROGEN', VOLUME_FRACTION(2)=1.88/

&REAC ID='Reaction-Mg', FUEL='MAGNESIUM', SPEC_ID_NU='AIR','MAGNESIUM','PRODUCTS', NU=-1.0,-1.0,1.0/

&SURF ID='FIRE-Mg', COLOR='RED', HRRPUA=5000.0, RAMP_Q='FIRE-Mg_RAMP_Q', TMP_FRONT=700.0, EMISSIVITY=0.7/
&RAMP ID='FIRE-Mg_RAMP_Q', T=0.0, F=0.0/
&RAMP ID='FIRE-Mg_RAMP_Q', T=30.0, F=1.0/
&RAMP ID='FIRE-Mg_RAMP_Q', T=60.0, F=1.0/
&RAMP ID='FIRE-Mg_RAMP_Q', T=70.0, F=0.0/

&OBST ID='Pan', XB=-0.2,0.2,-0.2,0.2,0.0,0.1, SURF_IDS='FIRE-Mg','INERT','INERT'/ 

&VENT SURF_ID='OPEN', MB='YMIN'/ 
&VENT SURF_ID='OPEN', MB='YMAX'/ 

&SLCF ID='Temp', QUANTITY='TEMPERATURE', VECTOR=.TRUE., PBY=0.0/

&BNDF QUANTITY='WALL TEMPERATURE'/

&TAIL /

Could you advise on how to resolve this issue? Are there any better options for this kind of simulation? Thank you.

BR
Aleks

**Comment by mcgratta** [comment]
I'll run the case.

**Comment by drjfloyd** [comment]
You have not specified any thermophysical properties for MgO. That could be having an impact.

**Comment by mcgratta** [comment]
The case failed for me too, but then I added
```
&CLIP MINIMUM_DENSITY=0.01 /
```
and it runs past the point of failure. I see temperatures reaching into the 5000 C range. I am not sure if that is to be expected or if there is something amiss with your stoichiometry and physical parameters.

**Comment by 19ap** [comment]
Thanks a lot for your suggestions. 
I’ve added the specific heat and conductivity values for MgO:

&SPEC ID='MAGNESIUM OXIDE', ... , SPECIFIC_HEAT=0.955, CONDUCTIVITY=45/

This adjustment has not prevented the simulation from stopping due to numerical instability, though it did delay the stoppage by 0.5 s.
At the same time, it has helped reduce the temperature at the flame level, so we no longer reach the 5000 C that Kevin had mentioned.

**Reply by drjfloyd** [reply]
I highly doubt that the conductivity of MgO vapor is 45 W/m/K.  Typical vapor conducitivies are on the order of 0.01 Wm/K

**Reply by 19ap** [reply]
I selected the conductivity of solid MgO here, as under typical conditions in air, the MgO formed during magnesium combustion rapidly cools and condenses into solid particles. Am I wrong?

**Reply by drjfloyd** [reply]
Please read the Technical Reference Guide and User's Guide. Your inputs are not how to define aerosol behavior in FDS.
